hopp/simulation/technologies/sites/site_shape_tools.py

```diff
@@ -0,0 +1,246 @@
+    """
+    site_length = np.sqrt(area_m2)
+    y1 = y0 + site_length
+    x1 = x0 + site_length
+    poly = box(x0,y0,x1,y1)
+    verts = [[x0,x0],[x1,y0],[x1,y1],[x0,y1]]
+    vertices = np.array([np.array(v) for v in verts])
+    return poly, vertices
+
+def make_rectangle(area_m2,aspect_ratio=1.5,x0=0.0,y0=0.0):
+    """Generate rectangle polygon shape of specified area.
+
+    Args:
+        area_m2 (float): area of shape in square meters.
+        aspect_ratio (float, Optional): ratio of width/height. Defaults to 1.5.
+            (width corresponds to x coordinates, height corresponds to y coordinates)
+        x0 (float, Optional): left-most x coordinate of the shape. Defaults to 0.0.
+        y0 (float, Optional): bottom-most x coordinate of the shape. Defaults to 0.0.
+
+    Returns:
+        2-element tuple containing
+
+        - **poly** (:obj:`shapely.geometry.Polygon`): site boundary polygon
+        - **vertices** (2D :obj:`numpy.ndarray`): vertices of site polygon. list of [x,y] coordinates in meters.        
+    """
+    height = np.sqrt(area_m2/aspect_ratio)
+    width = area_m2/height
+    x1 = x0 + width
+    y1 = y0 + height
+    poly = box(x0,y0,x1,y1)
+    verts = [[x0,x0],[x1,y0],[x1,y1],[x0,y1]]
+    vertices = np.array([np.array(v) for v in verts])
+    return poly,vertices
+
+def make_circle(area_m2,deg_diff = 10,x0=0.0,y0=0.0):
+    """Generate circle polygon shape of specified area.
+
+    Args:
+        area_m2 (float): area of shape in square meters.
+        deg_diff (float | int): difference in degrees for generating boundary. default to 10.
+            number of points generated is equal to ``360/deg_diff``
+        x0 (float, Optional): left-most x coordinate of the shape. Defaults to 0.0.
+        y0 (float, Optional): bottom-most x coordinate of the shape. Defaults to 0.0.
+
+    Returns:
+        2-element tuple containing
+
+        - **poly** (:obj:`shapely.geometry.Polygon`): site boundary polygon
+        - **vertices** (2D :obj:`numpy.ndarray`): vertices of site polygon. list of [x,y] coordinates in meters.
+    """
+    r = np.sqrt(area_m2/np.pi)
+    dx = np.deg2rad(deg_diff)
+    rads = np.arange(0,2*np.pi,dx)
+    x_coords = r*np.cos(rads)
+    y_coords = r*np.sin(rads)
+
+    if any(x<x0 for x in x_coords):
+        x_diff = [x0-x for x in x_coords if x<x0]
+        x_adj = max(x_diff)
+        x_points = [x+x_adj for x in x_coords]
+    else:
+        x_points = [x for x in x_coords]
+    if any(y<y0 for y in y_coords):
+        y_diff = [y0-y for y in y_coords if y<y0]
+        y_adj = max(y_diff)
+        y_points = [y+y_adj for y in y_coords]
+    else:
+        y_points = [y for y in y_coords]
+
+    coords = []
+    for x,y in zip(x_points,y_points):
+        coords.append([x,y])
+
+    poly = Polygon(coords)
+    vertices = np.array([np.array(v) for v in coords])
+    return poly, vertices
+
+def make_hexagon(area_m2,x0=0.0,y0=0.0):
```

Comment:
Can you 1) add spaces after the commas here and 2) update the list logic to utilize the numpy vectorization that I've demonstrated above?